src/main/kotlin/uk/gov/justice/digital/hmpps/educationandworkplanapi/app/database/jpa/LocationAuditingEntityListener.kt

```diff
@@ -0,0 +1,64 @@
+package uk.gov.justice.digital.hmpps.educationandworkplanapi.app.database.jpa
+
+import jakarta.persistence.PrePersist
+import jakarta.persistence.PreUpdate
+import java.lang.reflect.Field
+
+/**
+ * JPA Entity Listener that sets fields annotated with [CreatedLocation] or [LastModifiedLocation]] with the authenticated
+ * user's location, typically the prison in which they are working taken from their active caseload ID.
+ */
+class LocationAuditingEntityListener {
```

Comment:
This is exactly the same (copy and paste) of our existing `DisplayNameAuditingEntityListener`
Perhaps we can do something smarter by somehow combining the 2, but for the purpose of explaining the POC having it as a separate class probably makes it easier to understand 👍